versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5775 @@
+Boolean test_defined2 = defined(maybe_five_and_is)    # Evaluates to true
+Boolean test_is_none = maybe_five_but_is_not == None  # Evaluates to true
+Boolean test_not_none = maybe_five_but_is_not != None # Evaluates to false
+```
+
+For more details, see the section on [Optional Inputs with Defaults](#optional-inputs-with-defaults).
+
+#### Type Conversion
+
+WDL has some limited facilities for converting a value of one type to another type. Some of these are explicitly provided by [standard library](#standard-library) functions, while others are [implicit](#type-coercion). When converting between types, it is best to be explicit whenever possible, even if an implicit conversion is allowed.
+
+The execution engine is also responsible for converting (or "serializing") input values when constructing commands, as well as "deserializing" command outputs. For more information, see the [Command Section](#command-section) and the more extensive Appendix on [WDL Value Serialization and Deserialization](#appendix-a-wdl-value-serialization-and-deserialization).
+
+##### Type Coercion
+
+There are some pairs of WDL types for which there is an obvious, unambiguous conversion from one to the other. In these cases, WDL provides an automatic conversion (called "coercion") from one type to the other, such that a value of one type can be used anywhere the other type is expected.
+
+For example, file paths are always represented as strings, making the conversion from `String` to `File` obvious and unambiguous.
+
+```wdl
+String path = "/path/to/file"
+File f = path
+```
+
+The table below lists all globally valid coercions. The "target" type is the type being coerced to (this is often called the "left-hand side" or "LHS" of the coercion) and the "source" type is the type being coerced from (the "right-hand side" or "RHS").
+
+|Target Type |Source Type     |Notes/Constraints |
+|------------|----------------|------------------|
+|`File`|`String`||
+|`Float`|`Int`|May cause overflow error|
+|`Y?`|`X`|`X` must be coercible to `Y`|
+|`Array[Y]`|`Array[X]`|`X` must be coercible to `Y`|
+|`Map[X,Z]`|`Map[W,Y]`|`W` must be coercible to `X` and `Y` must be coercible to `Z`|
+|`Pair[X,Z]`|`Pair[W,Y]`|`W` must be coercible to `X` and `Y` must be coercible to `Z`|
+|`Object`|`Map[String,Y]`||
+|`Map[String,Y]`|`Object`|All object values must be coercible to `Y`|
+|`Struct`    |`Map[String,Y]`|`Map` keys must match `Struct` member names, and all `Struct` members types must be coercible from `Y`|
+|`Map[String,Y]`|`Struct`|All `Struct` members must be coercible to `Y`|
+|`Object`|`Struct`||
+|`Struct`|`Object`|`Object` keys must match `Struct` member names, and `Object` values must be coercible to `Struct` member types|
+
+Note that primitive types can always be converted to `String` using [string interpolation](#expression-placeholders-and-string-interpolation).
+
+```wdl
+Int i = 5
+String istring = "~{i}"
```

Comment:
This is the first time you've used `~{}`.  Just FYI.  It's an example that may not add much value as written in part b/c the reader doesn't technically know what this means yet, but I have a hard time giving you a better suggestion.